src/move_actions.c

```diff
@@ -306,7 +306,8 @@ bool8 YawnMoveAction(Entity * pokemon, Entity *target, Move *move, s32 param_4)
     return TRUE;
 }
 
-bool8 sub_80576F8(Entity * pokemon, Entity *target, Move *move, s32 param_4)
+// NOTE: Is there a better name for this?
+bool8 BasicSleepMoveAction(Entity * pokemon, Entity *target, Move *move, s32 param_4)
```

Comment:
Perhaps the "Basic" isn't needed. `SleepMoveAction` by itself conveys the same meaning of being a shared move action between similar sleep moves.

Reply:
If this is referring to rest etc, then Sleep is fine. If it's referring to forcing sleep (like hypnosis), then I suggest "CastSleep".
Same for "CastFreeze" and "CastBurn" etc if that's the situation

Reply:
It doesn't "force" the status conditions. It still has to roll the chance for a status condition. For example,
```c
bool8 BasicFireMoveAction(Entity *pokemon, Entity *target, Move *move, u32 param_4)
{
  bool8 flag;

  flag = FALSE;
  SendThawedMessage(pokemon, target);
  if (HandleDamagingMove(pokemon,target,move,0x100,param_4) != 0) {
    flag = TRUE;
    if(sub_805727C(pokemon, target, gUnknown_80F4DBA))
    {
        BurnedStatusTarget(pokemon, target, 0, FALSE);
    }
  }
  return flag;
}

```